### PR TITLE
feat(koogallery): add koogallery_assets data source

### PR DESCRIPTION
**Pull request by niuzhenguo**
**What this PR does / why we need it**:

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note

```

## PR Checklist

* [x] Tests added/passed.
* [x] Documentation updated.
* [x] Schema updated.

## Acceptance Steps Performed

```
make testacc TEST=./huaweicloud/services/acceptance/koogallery/ TESTARGS='-run=TestAccKooGalleryAssetsDataSource_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/koogallery/ -v -run=TestAccKooGalleryAssetsDataSource_basic -timeout 360m -parallel 4
=== RUN   TestAccKooGalleryAssetsDataSource_basic
=== PAUSE TestAccKooGalleryAssetsDataSource_basic
=== CONT  TestAccKooGalleryAssetsDataSource_basic
--- PASS: TestAccKooGalleryAssetsDataSource_basic (25.48s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/koogallery        25.513s
```

### CODE REVIEW

File: huaweicloud/services/koogallery/data_source_huaweicloud_koogallery_assets.go [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+	"context"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+
+	"github.com/chnsz/golangsdk/openstack/koogallery/v1/assets"
+
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/config"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/helper/hashcode"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/utils/fmtp"
```
Comment (ShiChangkuo) [inline huaweicloud/services/koogallery/data_source_huaweicloud_koogallery_assets.go]: it's better to use fmt package

File: huaweicloud/services/koogallery/data_source_huaweicloud_koogallery_assets.go [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+)
+
+func DataSourceKooGalleryAssets() *schema.Resource {
+	return &schema.Resource{
+		ReadContext: dataSourceKooGalleryAssetsRead,
+
```
Comment (ShiChangkuo) [inline huaweicloud/services/koogallery/data_source_huaweicloud_koogallery_assets.go:19]: it's better to add the Description field:
```
Description: "schema: Internal",
```

File: huaweicloud/services/koogallery/data_source_huaweicloud_koogallery_assets.go [hunk 3]
```diff
@@ -0,0 +1,192 @@
...
+		Schema: map[string]*schema.Schema{
+			"region": {
```
Comment (ShiChangkuo) [inline huaweicloud/services/koogallery/data_source_huaweicloud_koogallery_assets.go:21]: it seems that `region` is useless as the service is global.

Comment (niuzhenguo) [reply huaweicloud/services/koogallery/data_source_huaweicloud_koogallery_assets.go:21]: region here is a query parameter which is supported by the List API.

**Pull request merged**